sockeye/model.py

```diff
@@ -45,6 +47,8 @@ class ModelConfig(Config):
     :param config_loss: Loss configuration.
     :param lexical_bias: Use lexical biases.
     :param learn_lexical_bias: Learn lexical biases during training.
+    :param weight_tying: Enables weight tying if True.
```

Comment:
and I guess this is the right place to put them no? As it affects both encoder and decoder.

With respect to the name: we already have a parameter called weight_tying in the decoder config. I think it would be confusing to call it weight_tying in one place and weight_sharing in another. At the same time I don't want to necessarily break backwards compatibility.